### ISSUE TITLE
EDSC-909: Invalid value of echo_env gives 500 error

### ISSUE THREAD

**Issue by earthdata-tasktop** [nasa/edsc-search-test-config-3#250]
[http://edsc.dev/search/granules?echo_env=testbed1](http://edsc.dev/search/granules?echo_env=testbed1)

    NoMethodError (undefined method `[]' for nil:NilClass):
      lib/echo/client.rb:5:in `client_for_environment'
      app/controllers/application_controller.rb:21:in `echo_client'
      app/controllers/events_controller.rb:5:in `index'

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

[~bo.sun]and the problem was applicable to both?

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

h5.Steps to Verify

* Verify parameters cmr*env=prod, cmr*env=sit, cmr_env=uat load data from prod, sit and uat respectively
* Verify invalid value doesn't result in errors. i.e cmr_env=eqriur

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

[~kathleen*carr] echo*env was renamed to cmr_env.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

[~bo.sun]and the problem was applicable to both?

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

h5.Steps to Verify

* Verify parameters cmr*env=prod, cmr*env=sit, cmr_env=uat load data from prod, sit and uat respectively
* Verify invalid value doesn't result in errors. i.e cmr_env=eqriur

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

[~bo.sun]Hi Bo:  The ticket description talks about echo*env, but the verify instructions are for cmr*env.   Was the problem applicable to both?

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

[~kathleen*carr] echo*env was renamed to cmr_env.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

[~bo.sun]What happens if I enter an invalid value of cmr*env?  Does the query behave as if I had not entered cmr*env at all?

 I tried cmr*env=kathy, and did not get a 500 error (which is good), but I also did not get any indication that I had entered an invalid value for cmr*env (which I think is bad, since I won't know I made an error and may misinterpret my results).  

[https://search.uat.earthdata.nasa.gov/data/configure?cmr_env=kathy&amp;amp;p=!C446474-ORNL_DAAC&amp;amp;tl=557625600!4!!&amp;amp;q=USGS+FIFE&amp;amp;back=%2Fsearch%2Fproject](https://search.uat.earthdata.nasa.gov/data/configure?cmr_env=kathy&p=!C446474-ORNL_DAAC&tl=557625600!4!!&q=USGS+FIFE&back=%2Fsearch%2Fproject)

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

[~kathleen*carr], the 500 error only shows up when you put in an invalid environment parameter as the value of echo*env. This ticket fixed the error by ignoring invalid values and at the same time, renaming it to cmr_env to make it consistent across *earthdata.nasa.gov apps.

It is of course can be done to return an error on invalid env values. In my opinion, since the param will only be used internally for testing purpose, it is safe to just ignore the param and to show the default results set (i.e. if you are in UAT and send cmr_env=pprod, it will display uat collections.). On the other hand, given you are the one of few users who regularly use it, your take is also very important. So I leave this to [~pquinn] to decide. ;)

**Comment by earthdata-tasktop** [comment]
(Comment from Dana Shum):

Migrating "Internal Verified" tickets to "Completed".

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

Talked with Patrick on 2/17/16, and the consensus is that because of the small number of users of the cmr*env function, it is not worth spending the time to return an error on invalid values of cmr*env.   cmr*env=ops and cmr*env=prod will both be supported in the future, which will alleviate one source of possible error.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

Verified that the new parameter cmr_env (= prod, uat, sit) point the query to the corresponding environment.

If I enter an invalid value for cmr*env, e.g.,  cmr*env=kathy, I do not get a 500 error (which is good), but I also do not get any indication that I had entered an invalid value for cmr*env.   The behavior was to ignore the cmr*env parameter altogether and just use the current environment.

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

h5.Steps to Verify

* Verify parameters cmr*env=prod, cmr*env=sit, cmr_env=uat load data from prod, sit and uat respectively
* Verify invalid value doesn't result in errors. i.e cmr_env=eqriur

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

[~bo.sun]Hi Bo:  The ticket description talks about echo*env, but the verify instructions are for cmr*env.   Was the problem applicable to both?

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

[~kathleen*carr] echo*env was renamed to cmr_env.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

[~bo.sun]What happens if I enter an invalid value of cmr*env?  Does the query behave as if I had not entered cmr*env at all?

 I tried cmr*env=kathy, and did not get a 500 error (which is good), but I also did not get any indication that I had entered an invalid value for cmr*env (which I think is bad, since I won't know I made an error and may misinterpret my results).  

[https://search.uat.earthdata.nasa.gov/data/configure?cmr_env=kathy&amp;amp;p=!C446474-ORNL_DAAC&amp;amp;tl=557625600!4!!&amp;amp;q=USGS+FIFE&amp;amp;back=%2Fsearch%2Fproject](https://search.uat.earthdata.nasa.gov/data/configure?cmr_env=kathy&p=!C446474-ORNL_DAAC&tl=557625600!4!!&q=USGS+FIFE&back=%2Fsearch%2Fproject)

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

[~kathleen*carr], the 500 error only shows up when you put in an invalid environment parameter as the value of echo*env. This ticket fixed the error by ignoring invalid values and at the same time, renaming it to cmr_env to make it consistent across *earthdata.nasa.gov apps.

It is of course can be done to return an error on invalid env values. In my opinion, since the param will only be used internally for testing purpose, it is safe to just ignore the param and to show the default results set (i.e. if you are in UAT and send cmr_env=pprod, it will display uat collections.). On the other hand, given you are the one of few users who regularly use it, your take is also very important. So I leave this to [~pquinn] to decide. ;)

**Comment by earthdata-tasktop** [comment]
(Comment from Dana Shum):

Migrating "Internal Verified" tickets to "Completed".

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

Talked with Patrick on 2/17/16, and the consensus is that because of the small number of users of the cmr*env function, it is not worth spending the time to return an error on invalid values of cmr*env.   cmr*env=ops and cmr*env=prod will both be supported in the future, which will alleviate one source of possible error.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

Verified that the new parameter cmr_env (= prod, uat, sit) point the query to the corresponding environment.

If I enter an invalid value for cmr*env, e.g.,  cmr*env=kathy, I do not get a 500 error (which is good), but I also do not get any indication that I had entered an invalid value for cmr*env.   The behavior was to ignore the cmr*env parameter altogether and just use the current environment.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

[~bo.sun]Hi Bo:  The ticket description talks about echo*env, but the verify instructions are for cmr*env.   Was the problem applicable to both?

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

[~kathleen*carr] echo*env was renamed to cmr_env.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

[~bo.sun]What happens if I enter an invalid value of cmr*env?  Does the query behave as if I had not entered cmr*env at all?

 I tried cmr*env=kathy, and did not get a 500 error (which is good), but I also did not get any indication that I had entered an invalid value for cmr*env (which I think is bad, since I won't know I made an error and may misinterpret my results).  

[https://search.uat.earthdata.nasa.gov/data/configure?cmr_env=kathy&amp;amp;p=!C446474-ORNL_DAAC&amp;amp;tl=557625600!4!!&amp;amp;q=USGS+FIFE&amp;amp;back=%2Fsearch%2Fproject](https://search.uat.earthdata.nasa.gov/data/configure?cmr_env=kathy&p=!C446474-ORNL_DAAC&tl=557625600!4!!&q=USGS+FIFE&back=%2Fsearch%2Fproject)

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

[~kathleen*carr], the 500 error only shows up when you put in an invalid environment parameter as the value of echo*env. This ticket fixed the error by ignoring invalid values and at the same time, renaming it to cmr_env to make it consistent across *earthdata.nasa.gov apps.

It is of course can be done to return an error on invalid env values. In my opinion, since the param will only be used internally for testing purpose, it is safe to just ignore the param and to show the default results set (i.e. if you are in UAT and send cmr_env=pprod, it will display uat collections.). On the other hand, given you are the one of few users who regularly use it, your take is also very important. So I leave this to [~pquinn] to decide. ;)

**Comment by earthdata-tasktop** [comment]
(Comment from Dana Shum):

Migrating "Internal Verified" tickets to "Completed".

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

Talked with Patrick on 2/17/16, and the consensus is that because of the small number of users of the cmr*env function, it is not worth spending the time to return an error on invalid values of cmr*env.   cmr*env=ops and cmr*env=prod will both be supported in the future, which will alleviate one source of possible error.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

Verified that the new parameter cmr_env (= prod, uat, sit) point the query to the corresponding environment.

If I enter an invalid value for cmr*env, e.g.,  cmr*env=kathy, I do not get a 500 error (which is good), but I also do not get any indication that I had entered an invalid value for cmr*env.   The behavior was to ignore the cmr*env parameter altogether and just use the current environment.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

[~bo.sun]What happens if I enter an invalid value of cmr*env?  Does the query behave as if I had not entered cmr*env at all?

 I tried cmr*env=kathy, and did not get a 500 error (which is good), but I also did not get any indication that I had entered an invalid value for cmr*env (which I think is bad, since I won't know I made an error and may misinterpret my results).  

[https://search.uat.earthdata.nasa.gov/data/configure?cmr_env=kathy&amp;amp;p=!C446474-ORNL_DAAC&amp;amp;tl=557625600!4!!&amp;amp;q=USGS+FIFE&amp;amp;back=%2Fsearch%2Fproject](https://search.uat.earthdata.nasa.gov/data/configure?cmr_env=kathy&p=!C446474-ORNL_DAAC&tl=557625600!4!!&q=USGS+FIFE&back=%2Fsearch%2Fproject)

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

[~kathleen*carr], the 500 error only shows up when you put in an invalid environment parameter as the value of echo*env. This ticket fixed the error by ignoring invalid values and at the same time, renaming it to cmr_env to make it consistent across *earthdata.nasa.gov apps.

It is of course can be done to return an error on invalid env values. In my opinion, since the param will only be used internally for testing purpose, it is safe to just ignore the param and to show the default results set (i.e. if you are in UAT and send cmr_env=pprod, it will display uat collections.). On the other hand, given you are the one of few users who regularly use it, your take is also very important. So I leave this to [~pquinn] to decide. ;)

**Comment by earthdata-tasktop** [comment]
(Comment from Dana Shum):

Migrating "Internal Verified" tickets to "Completed".

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

Talked with Patrick on 2/17/16, and the consensus is that because of the small number of users of the cmr*env function, it is not worth spending the time to return an error on invalid values of cmr*env.   cmr*env=ops and cmr*env=prod will both be supported in the future, which will alleviate one source of possible error.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

Verified that the new parameter cmr_env (= prod, uat, sit) point the query to the corresponding environment.

If I enter an invalid value for cmr*env, e.g.,  cmr*env=kathy, I do not get a 500 error (which is good), but I also do not get any indication that I had entered an invalid value for cmr*env.   The behavior was to ignore the cmr*env parameter altogether and just use the current environment.

**Comment by earthdata-tasktop** [comment]
(Comment from Bo Sun):

[~kathleen*carr], the 500 error only shows up when you put in an invalid environment parameter as the value of echo*env. This ticket fixed the error by ignoring invalid values and at the same time, renaming it to cmr_env to make it consistent across *earthdata.nasa.gov apps.

It is of course can be done to return an error on invalid env values. In my opinion, since the param will only be used internally for testing purpose, it is safe to just ignore the param and to show the default results set (i.e. if you are in UAT and send cmr_env=pprod, it will display uat collections.). On the other hand, given you are the one of few users who regularly use it, your take is also very important. So I leave this to [~pquinn] to decide. ;)

**Comment by earthdata-tasktop** [comment]
(Comment from Dana Shum):

Migrating "Internal Verified" tickets to "Completed".

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

Talked with Patrick on 2/17/16, and the consensus is that because of the small number of users of the cmr*env function, it is not worth spending the time to return an error on invalid values of cmr*env.   cmr*env=ops and cmr*env=prod will both be supported in the future, which will alleviate one source of possible error.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

Verified that the new parameter cmr_env (= prod, uat, sit) point the query to the corresponding environment.

If I enter an invalid value for cmr*env, e.g.,  cmr*env=kathy, I do not get a 500 error (which is good), but I also do not get any indication that I had entered an invalid value for cmr*env.   The behavior was to ignore the cmr*env parameter altogether and just use the current environment.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

Talked with Patrick on 2/17/16, and the consensus is that because of the small number of users of the cmr*env function, it is not worth spending the time to return an error on invalid values of cmr*env.   cmr*env=ops and cmr*env=prod will both be supported in the future, which will alleviate one source of possible error.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

Verified that the new parameter cmr_env (= prod, uat, sit) point the query to the corresponding environment.

If I enter an invalid value for cmr*env, e.g.,  cmr*env=kathy, I do not get a 500 error (which is good), but I also do not get any indication that I had entered an invalid value for cmr*env.   The behavior was to ignore the cmr*env parameter altogether and just use the current environment.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

Verified that the new parameter cmr_env (= prod, uat, sit) point the query to the corresponding environment.

If I enter an invalid value for cmr*env, e.g.,  cmr*env=kathy, I do not get a 500 error (which is good), but I also do not get any indication that I had entered an invalid value for cmr*env.   The behavior was to ignore the cmr*env parameter altogether and just use the current environment.

**Comment by earthdata-tasktop** [comment]
(Comment from Dana Shum):

Migrating "Internal Verified" tickets to "Completed".

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

Talked with Patrick on 2/17/16, and the consensus is that because of the small number of users of the cmr*env function, it is not worth spending the time to return an error on invalid values of cmr*env.   cmr*env=ops and cmr*env=prod will both be supported in the future, which will alleviate one source of possible error.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

Verified that the new parameter cmr_env (= prod, uat, sit) point the query to the corresponding environment.

If I enter an invalid value for cmr*env, e.g.,  cmr*env=kathy, I do not get a 500 error (which is good), but I also do not get any indication that I had entered an invalid value for cmr*env.   The behavior was to ignore the cmr*env parameter altogether and just use the current environment.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

Talked with Patrick on 2/17/16, and the consensus is that because of the small number of users of the cmr*env function, it is not worth spending the time to return an error on invalid values of cmr*env.   cmr*env=ops and cmr*env=prod will both be supported in the future, which will alleviate one source of possible error.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

Verified that the new parameter cmr_env (= prod, uat, sit) point the query to the corresponding environment.

If I enter an invalid value for cmr*env, e.g.,  cmr*env=kathy, I do not get a 500 error (which is good), but I also do not get any indication that I had entered an invalid value for cmr*env.   The behavior was to ignore the cmr*env parameter altogether and just use the current environment.

**Comment by earthdata-tasktop** [comment]
(Comment from Kathleen Carr):

Verified that the new parameter cmr_env (= prod, uat, sit) point the query to the corresponding environment.

If I enter an invalid value for cmr*env, e.g.,  cmr*env=kathy, I do not get a 500 error (which is good), but I also do not get any indication that I had entered an invalid value for cmr*env.   The behavior was to ignore the cmr*env parameter altogether and just use the current environment.
